src/Magnum/DartIntegration/ConvertShapeNode.cpp

```diff
@@ -0,0 +1,193 @@
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include <dart/dynamics/BoxShape.hpp>
+#include <dart/dynamics/CapsuleShape.hpp>
+#include <dart/dynamics/CylinderShape.hpp>
+#include <dart/dynamics/EllipsoidShape.hpp>
+#include <dart/dynamics/MeshShape.hpp>
+#include <dart/dynamics/ShapeNode.hpp>
+#include <dart/dynamics/SphereShape.hpp>
+
+#include <Corrade/PluginManager/Manager.h>
+#include <Corrade/Utility/Directory.h>
+
+#include <Magnum/MeshTools/Transform.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Primitives/Capsule.h>
+#include <Magnum/Primitives/Cube.h>
+#include <Magnum/Primitives/Cylinder.h>
+#include <Magnum/Primitives/Icosphere.h>
+#include <Magnum/TextureFormat.h>
+#include <Magnum/Trade/AbstractImporter.h>
+#include <Magnum/Trade/ImageData.h>
+#include <Magnum/Trade/MeshData3D.h>
+#include <Magnum/Trade/PhongMaterialData.h>
+#include <Magnum/Trade/TextureData.h>
+
+#include <Magnum/DartIntegration/ConvertShapeNode.h>
+
+namespace Magnum { namespace DartIntegration {
+
+Corrade::Containers::Optional<VisualData> convertShapeNode(dart::dynamics::ShapeNode* shapeNode) {
+    if (!shapeNode)
+        return Corrade::Containers::NullOpt;
+    // Types from DART are:
+    //    - BoxShape
+    //    - CapsuleShape
+    //    x- ConeShape
+    //    - CylinderShape
+    //    - EllipsoidShape
+    //    x- LineSegmentShape
+    //    - MeshShape
+    //    x- MultiSphereConvexHullShape
+    //    x- PlaneShape (this is infite plane with normal)
+    //    x- SoftMeshShape
+    //    - SphereShape
+    auto shape = shapeNode->getShape();
+
+    VisualData data;
+
+    // Get material information -- we ignore the alpha value
+    // Note that if this is a MeshShape we will ignore this material
+    Eigen::Vector4d col = shapeNode->getVisualAspect()->getRGBA();
+
+    // default shininess to 80.f
+    data._matData = new Trade::PhongMaterialData{Trade::PhongMaterialData::Flags{}, 80.f};
+    data._matData->ambientColor() = Color3{Float(col(0)), Float(col(1)), Float(col(2))};
+    data._matData->diffuseColor() = data._matData->ambientColor();
+    data._matData->specularColor() = data._matData->ambientColor();
+
+    if (shape->getType() == "BoxShape") {
+        auto boxShape = std::dynamic_pointer_cast<dart::dynamics::BoxShape>(shape);
```

Comment:
I think `dynamic_pointer_cast` is exactly what we need (and what the Dart developers suggest to use in this case), because a run-time check is performed: https://stackoverflow.com/questions/28002/regular-cast-vs-static-cast-vs-dynamic-cast

Btw, I could/should check if the pointer is valid!

Reply:
Wait, so it can happen that `getType()` returns `"BoxShape"`, but the type is *not* `BoxShape`? I hope not! :fearful:

Reply:
It should never happen according to Dart docs. So I will not check! :)

Reply:
So then you can also use `std::static_pointer_cast`, as dynamic cast is only a type check on top of it :)